src/Raven.Client/Documents/Session/DocumentQuery.cs

```diff
@@ -774,6 +844,26 @@ IDocumentQuery<TResult> IDocumentQuery<T>.OfType<TResult>()
             return this;
         }
 
+        IGraphQuery<T> IGraphQuery<T>.With<TOther>(string alias, IRavenQueryable<TOther> query)
+        {
+            //TODO:Find a proper way of do this extraction
+            var queryInspector = (RavenQueryInspector<TOther>)query;
+            var docQuery = (DocumentQuery<TOther>)queryInspector.GetDocumentQuery(x => x.ParameterPrefix = $"w{WithTokens.Count}p");
+            if (docQuery.SelectTokens?.Count > 0)
+            {
+                throw new NotSupportedException($"Select is not premitted in a 'With' clause in query:{docQuery}");
+            }
+
+
+            foreach (var keyValue in docQuery.QueryParameters)
+            {
+                QueryParameters.Add(keyValue.Key, keyValue.Value);
```

Comment:
Maybe we should use `AddParameter` method. I do not remember if we are doing some validation there or not.